From a2e043603e2a572d4a09af311cc3d1261d21c2df Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Mon, 22 Feb 2021 17:42:50 +0800
Subject: [PATCH] drm/rockchip: vop2: Fix bg_ovl_dly

It should be prev_vp, vp is a fixed port in this function.

Change-Id: I7c5b1be4ecebbf6a3aee7766a0f8ecbcbf4951a1
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 971015641a0d..9b082d40ce61 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -4310,9 +4310,9 @@ static void vop2_setup_layer_mixer_for_vp(struct vop2_video_port *vp,
 			port_mux = used_layers - 1;
 
 		if (port_mux > vop2_data->nr_mixers)
-			vp->bg_ovl_dly = 0;
+			prev_vp->bg_ovl_dly = 0;
 		else
-			vp->bg_ovl_dly = (vop2_data->nr_mixers - port_mux) << 1;
+			prev_vp->bg_ovl_dly = (vop2_data->nr_mixers - port_mux) << 1;
 		VOP_MODULE_SET(vop2, prev_vp, port_mux, port_mux);
 	}
 
-- 
2.35.3

